querybook/webapp/redux/dataTableSearch/types.ts

```diff
@@ -89,11 +134,21 @@ export type DataTableSearchAction =
     | IDataTableSearchFilterUpdateAction
     | IDataTableSearchResetFilterAction
     | IDataTableSearchMoreAction
-    | IDataTableSearchSelectMetastoreAction;
+    | IDataTableSearchSelectMetastoreAction
+    | ISchemaSearchMoreAction
+    | ISchemaSearchResultAction
+    | ISearchTableBySchemaAction
+    | ISearchTableBySchemaResultAction
+    | ISchemaSearchFailedAction
+    | ISearchTableBySchemaFailedAction;
 
 export interface IDataTableSearchPaginationState {
     results: ITableSearchResult[];
     count: number;
+    schemas: {
```

Comment:
use dictionary to speed up indexing, I would recommend the following shape:
```
schemaIds: number[];
schemaResultById: Record<number, ISchemaTableSearch>;
```

there is no need for count (see above for reasoning) but you can have a state to track whether or no there are more results